include/core/JamomaSample.h

```diff
@@ -244,7 +254,92 @@ namespace Jamoma {
 			for (int i=0; i<frameCount; ++i)
 				values[i] = vector[i];
 		}
-		
+        
+        
+        /** Duplicate values at the beginning and end to provide padding for interpolation.
+            @param  paddingAmount   Number of frames to add to both beginning & end of SampleBundle. Total number of frames added to SampleBundle will be twice this amount.
+         */
+        void applySamplePadding(int paddingAmount)
+        {
+            
+            // ensure that we are adding at least one sample, AND not more than half SampleBundle frames
+            assert(paddingAmount > 0 && paddingAmount < mFrameCount/2);
+            
+            // make a copy
+            SampleBundleData newPaddedChannels = mChannels;
+            
+            for (auto& channel : newPaddedChannels) {
+                
+                // set up temporary vectors for padding samples
+                SampleVector firstSamples(paddingAmount,0.0);
+                SampleVector lastSamples(paddingAmount,0.0);
+                
+                // copy samples from the beginning and end of SampleBundle
+                for (int i=0; i<paddingAmount; i++) {
+                    lastSamples[i] = channel[i];
+                    firstSamples[i] = channel[ (frameCount()-paddingAmount) + i ];
+                }
+                
+                // add them to the beginning
+                channel.insert(channel.begin(),
+                               firstSamples.begin(),
+                               firstSamples.end());
+                
+                // add them to the end
+                channel.insert(channel.end(),
+                               lastSamples.begin(),
+                               lastSamples.end());
+                
+            }
+            
+            // save back to SampleBundleData
+            mChannels = newPaddedChannels;
+            
```

Comment:
We made a copy up above (see my previous comment).  But we shouldn't need to copy here.  We should be able to **move** instead.  Might have to use std::move here.  @nwolek , please feel free to make me a ticket to investigate this.  Even if we have to copy up above, maybe we can cut our trouble down here somewhat?